www/js/lib/uiUtil.js

```diff
@@ -227,12 +228,92 @@ define([], function() {
+        var kiwixJSSheet = doc ? doc.getElementById('kiwixJSTheme') || null : null;
+        var appTheme = theme.replace(/_.*$/, '');
+        var contentTheme = theme.replace(/^[^_]*/, '');
+        var oldAppTheme = oldTheme.replace(/_.*$/, '');
+        var oldContentTheme = oldTheme.replace(/^[^_]*/, '');
+        // Remove oldAppTheme if necessary
+        if (oldAppTheme && oldAppTheme !== appTheme) htmlEl.classList.remove(oldAppTheme);
+        // Apply new appTheme (NB it will not be added twice if it's already there)
+        if (appTheme) htmlEl.classList.add(appTheme);
+        // Embed a reference to applied theme, so we can remove it generically in the future
+        htmlEl.dataset.theme = theme;
+        // Hide any previously displayed help
+        var oldHelp = document.getElementById(oldTheme + '-help');
+        if (oldHelp) oldHelp.style.display = 'none';
+        // Show any specific help for selected contentTheme
+        var help = document.getElementById(theme + '-help');
+        if (help) help.style.display = 'block';
+        
+        // If there is no ContentTheme or we are applying a different ContentTheme, remove any previously applied ContentTheme
+        if (oldContentTheme && oldContentTheme !== contentTheme) {
+            iframe.classList.remove(oldContentTheme);
+            if (kiwixJSSheet) {
+                kiwixJSSheet.disabled = true;
+                kiwixJSSheet.parentNode.removeChild(kiwixJSSheet);
+            }
+        }
+        // Apply the requested ContentTheme (if not already attached)
+        if (contentTheme && (!kiwixJSSheet || !~kiwixJSSheet.href.search(contentTheme))) {
+            iframe.classList.add(contentTheme);
+            // Use an absolute reference becuase Service Worker needs this
+            var prefix = document.location.href.replace(/index\.html(?:$|[#?].*$)/, '');
+            if (doc) {
+                var link = doc.createElement('link');
+                link.setAttribute('id', 'kiwixJSTheme');
+                link.setAttribute('rel', 'stylesheet');
+                link.setAttribute('type', 'text/css');
+                link.setAttribute('href', prefix + 'css/kiwixJS' + contentTheme + '.css');
+                doc.head.appendChild(link);
+            }
+        }
+        // If we are in Config and a real document has been loaded already, expose return link so user can see the result of the change
+        if (/active/.test(document.getElementById('liConfigureNav').classList) &&
+            !/Placeholder\sfor\sinjecting\san\sarticle/.test(doc.title)) {
+            showReturnLink();
+        }
+    }
+
+    // Displays the return link and handles click event. Called by applyAppTheme()
+    function showReturnLink() {
+        var viewArticle = document.getElementById('viewArticle');
+        viewArticle.style.display = 'block';
+        viewArticle.addEventListener('click', function(e) {
+            e.preventDefault();
```

Comment:
@mossroy Once #579 is merged, I'll rebase this PR and can then use the new functions (I think) for adding the swipe.
This is a part of #127, but doesn't reload the article, simply doesn't wipe it from the Home div (there's no need to reload to apply the theme, because it is added using DOM methods).
I still think the return link is useful to the user, because it orients them after what could be a somewhat startling style change, especially for new users. The link is removed as soon as it is used. I need to ensure it is also removed if a different article is loaded (I'll add a line to hide it to the `applyAppTheme` function).

Reply:
Regarding implementation of #127, I think that really needs #523 to be complete first, otherwise I fear we'll be coding the feature twice, once for the current UI and again for the more compact UI.

Reply:
PS I'd be happy to remove this in the future if it turns out that #127 elegantly supersedes it.